### PR TITLE
Add exponential backoff

### PR DESCRIPTION
**Pull request by fdr**
On the basis of previous experience, it's been bothering me for some
time that strands that crash or fail to terminate will be scheduled
often, and in fact, more often than code that takes even a short nap,
since "schedule" is set to "now()", to implement a round-robin
approach (bc4911fae065b19d9e1175e557b826bab8ca28a0).

This patch implements an exponential backoff algorithm with jitter, so
that strands that do not raise `FlowControl` exceptions per normal
will receive less execution time.

The most important, subtle aspect of this patch is how failure is
*assumed* in the initial lease-taking `UPDATE` statement, and only on
successful executions is the `try` count re-set to zero.  The reason
for this is that the code executed by Strand is capable of
non-termination.  Thus, the code is written such that failure is
assumed, and effectively backed-out when a `FlowControl` exception is
raised, and schedule is overwritten a second time.  The previous,
simpler, always-round-robin-at-least code from before was designed on
a similar principle, updating first, then updating again.

However, this forward-dating and back-dating of both `try` and
`schedule` can piggyback on other queries and can be HOT updated in
Postgres, so the anticipated added cost is minuscule.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -71,3 +71,6 @@ RSpec/InstanceVariable:
 RSpec/SpecFilePathFormat:
   Exclude:
     - 'spec/routes/**/*.rb'
+
+Sequel/ColumnDefault:
+  Enabled: false
```
Comment (byucesoy) [inline .rubocop.yml:76]: Why do we need this? Removing this and running rubocop on my local didn't generate any errors.

File: model/strand.rb [hunk 2]
```diff
@@ -38,7 +38,7 @@ def self.prog_verify(prog)
   def self.lease(id)
     affected = DB[<<SQL, id].first
 UPDATE strand
-SET lease = now() + '120 seconds', schedule = now()
+SET lease = now() + '120 seconds', try = try + 1, schedule = (now() + least(2 ^ least(try, 20) * random(), 600) * '1 second'::interval)
```
Comment (byucesoy) [inline model/strand.rb:41]: Shouldn't we put random() to outside of least?
```
now() + least(2 ^ least(try, 20), 600) * random() * '1 second'::interval
```
Currently when try is high, it is very likely that 600 will be picked, causing multiple jobs to be scheduled at the same time. For example if try is 10 we will pick 600 approximately 40% of the time, for 11 it is 70%, for 12 it is 85% and so on.

Comment (fdr) [reply model/strand.rb:41]: Good idea

File: .rubocop.yml (hunk 1 above)
Comment (fdr) [reply .rubocop.yml:76]: Huh, it definitely breaks for me with the migration I added.

Comment (fdr) [reply .rubocop.yml:76]: .....I wonder if adding `null: false` "fixed" it

**Review by byucesoy — APPROVED** [review]

**Pull request merged**